### PR TITLE
Link check command feature

### PR DESCRIPTION
**Pull request by alexbn71**
Currently there is no method to check if a node is really still joined to TTN or if the gateway/s in range, is/are still up and running.
The "join bit" of "mac get status" response is not reliable... indeed if the only gateway in range is stopped this bit continues to return 1 (joined). Nothing strange since the rule is fire-and-forget.

A work around could then use "mac set linkchk nnn". 

_This function sets the time interval for the link check process to be triggered periodically. A <value> of ‘0’ will disable the link check process. When the time interval expires, the next application packet that will be sent to the server will include a link check MAC command. Refer to the LoRaWAN Specification V1.0 document for more information on the link check configuration._

Thanks to this command is it possible then get the number of gateways in range and the demodulation margin in dB returned on the last check.

NB: to not disobey at the TTN Fair Access Policy is in charge of those who use it to make the number of downlink stay within the limits allowed. The max value for "nnn" is 65.535 which corresponds to an interval of 18 hours about.

I've tested this method on my end node and gateway and it seems realiable, see this piece of log:

```
-- LINK CHECK ENABLED
-- GATEWAYS 1 - MARGIN 16
Sending: mac tx uncnf 1 08011D0000F041250000AC422D0000C242
Successful transmission
Sending: sys sleep 60000
-- LINK CHECK ENABLED
-- GATEWAYS 0 - MARGIN 255
Sending: mac tx uncnf 1 08011D0000F041250000AC422D0000C242
Successful transmission
Sending: sys sleep 60000
-- LINK CHECK ENABLED
-- GATEWAYS 0 - MARGIN 255
Sending: mac tx uncnf 1 08011D0000F041250000AC422D0000C242
Successful transmission
Sending: sys sleep 60000
-- LINK CHECK ENABLED
-- GATEWAYS 0 - MARGIN 255
Sending: mac tx uncnf 1 08011D0000F041250000AC422D0000C242
Successful transmission
Sending: sys sleep 60000
-- LINK CHECK ENABLED
-- GATEWAYS 0 - MARGIN 255
Sending: mac tx uncnf 1 08011D0000F041250000AC422D0000C242
Successful transmission
Sending: sys sleep 60000
-- LINK CHECK ENABLED
-- GATEWAYS 1 - MARGIN 17
Sending: mac tx uncnf 1 08011D0000F041250000AC422D0000C242
Successful transmission
Sending: sys sleep 60000
```

**Explanation: after the classic OTAA the values returned was "-- GATEWAYS 1 - MARGIN 16" then I stopped the "ttn-pkt-fwd" service and it became "-- GATEWAYS 0 - MARGIN 255"... then restarting again "ttn-pkt-fwd" returned to be "GATEWAYS 1 - MARGIN 17".**

### CODE REVIEW

**Comment by jpmeijers** [comment]
Seems like a useful feature to me. Can someone in range of multiple gateways perhaps test this feature?

File: src/TheThingsNetwork.cpp [hunk 1]
```diff
@@ -963,3 +967,32 @@ void TheThingsNetwork::wake()
 { 
   autoBaud();
 }
+
+void TheThingsNetwork::linkcheck(uint16_t seconds)
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: `linkCheck`

Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: `seconds` means interval here, right?

File: src/TheThingsNetwork.cpp [hunk 2]
```diff
@@ -963,3 +967,32 @@ void TheThingsNetwork::wake()
...
+  sendCommand(MAC_GET_SET_TABLE, MAC_LINKCHK, true);
+
+  sprintf(buffer, "%u", seconds);
+  modemStream->write(buffer);
+  modemStream->write(SEND_MSG);
+  debugPrintLn(buffer);
+  waitForOk();  
+}
+
+uint8_t TheThingsNetwork::linkcheckGateways()
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: `getLinkCheckGateways()`

File: src/TheThingsNetwork.cpp [hunk 3]
```diff
@@ -963,3 +967,32 @@ void TheThingsNetwork::wake()
...
+{
+  readResponse(MAC_TABLE, MAC_GET_SET_TABLE, MAC_GWNB, buffer, sizeof(buffer));
+
+  return strtol(buffer, NULL, 10);
+}
+
+uint8_t TheThingsNetwork::linkcheckMargin()
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: `getLinkCheckMargin()`

File: src/TheThingsNetwork.cpp [hunk 4]
```diff
@@ -963,3 +967,32 @@ void TheThingsNetwork::wake()
...
+{
+  readResponse(MAC_TABLE, MAC_GET_SET_TABLE, MAC_MRGN, buffer, sizeof(buffer));
+
+  return strtol(buffer, NULL, 10);
+}
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: Newline

File: src/TheThingsNetwork.cpp (hunk 1 above)
Comment (alexbn71) [inline src/TheThingsNetwork.cpp]: Yes it is, no problem if you prefer to use the keyword interval.

Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: No it's ok like this!

**Comment by alexbn71** [comment]
Hi Johan, all done...

**Comment by johanstokking** [comment]
Released in https://github.com/TheThingsNetwork/arduino-device-lib/releases/tag/v2.5.6

**Pull request merged**